src/test/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitCheckerTest.java

```diff
@@ -241,29 +243,37 @@ public void ThrottleOnOverTest() throws Exception {
         // check rate limit to hit the first 11 scenarios because the throttle (add more here)
         // does not happen until under buffer
         for (int i = 0; i < 11; i++) {
-            ApiRateLimitChecker.ThrottleOnOver.checkApiRateLimit(listener, github);
+            assertFalse(ApiRateLimitChecker.ThrottleOnOver.checkApiRateLimitOnce(listener, github, 0));
         }
 
         //should be no output
         assertEquals(0, countOfOutputLines(m -> m.matches(".*[sS]leeping.*")));
 
         assertEquals(11, getRequestCount(githubApi));
 
-        // check rate limit to hit the next 10 scenarios
-        ApiRateLimitChecker.ThrottleOnOver.checkApiRateLimit(listener, github);
+        // check rate limit to hit the next 9 scenarios
+        for (int i = 0; i < 10; i++) {
+            assertTrue(ApiRateLimitChecker.ThrottleOnOver.checkApiRateLimitOnce(listener, github, i));
```

Comment:
By removing "rechecking" we remove the ability of tests to shortcut long wait loops by providing updated data. 
Instead we not do these waits manually for only a specified number of iterations in testing.